File: 配套资源/配套代码/程序设计竞赛专题挑战教程-代码/第4章蓝桥杯2022省赛求和.cpp
//lanqiaoOJ  2080

#include<iostream>
using namespace std;
#define ll long long
const int N = 200000 + 100;
ll n, a[N], sum[N];
int main()
{
    scanf("%lld", &n);
    for(int i = 1; i <= n; i++)
    {
        scanf("%lld", a + i);
        sum[i] = sum[i - 1] + a[i];
    }
    ll ans = 0;
    for(int i = 1; i <= n; i++)
        ans += a[i] * (sum[n] - sum[i]);

    printf("%lld", ans);
    return 0;
}

File: 配套资源/配套代码/程序设计竞赛专题挑战教程-代码/第4章蓝桥杯2022省赛求和.py
# lanqiaoOJ 2080
# 2022省赛 求和
n = int(input())
a=[int(i) for i in input().split()]          #注意：读入a时，范围是a[0]~a[n-1]
sum = [0] * n                                #定义前缀和
sum[0] = a[0]
for i in range(1,n): sum[i] = a[i]+sum[i-1]  #计算前缀和sum[]
s = 0
for i in range(0,n-1):  s += sum[i]*a[i+1]   #计算和s
print(s)
